File: rad_anastasiya/bridge/src/com/universiry/bridge/shape/Rectangle.java
package com.universiry.bridge.shape;

import com.universiry.bridge.color.IColor;

public class Rectangle extends Shape {
    protected final double first;
    protected final double second;

    public Rectangle(final double first, final double second, final IColor IColor) {
        super(IColor);
        this.first = first;
        this.second = second;
    }

    @Override
    public double square() {
        return first * second;
    }

    @Override
    public double perimeter() {
        return 2 * (first + second);
    }

    @Override
    public String toString() {
        return "Rectangle{" +
                "first: " + first +
                ", second: " + second +
                ", color: " + IColor.name()
                + ", perimeter: " + perimeter()
                + ", square: " + square()
                + '}';
    }
}

File: kulikovskiy_georgiy/Bridge/Shape/Rectangle.cs
namespace Bridge
{
    public class Rectangle : Shape
    {
        public double SideTwo { get; set; }

        public Rectangle(double sideOne, double sideTwo, IColor color)
            : base(sideOne, color)
        {
            SideTwo = sideTwo;
        }

        public override double Area()
        {
            return SideOne * SideTwo;
        }

        public override double Perimeter()
        {
            return 2 * (SideOne + SideTwo);
        }

        public override string ToString()
        {
            return $"Прямокутник: "
                + $"сторона one: {SideOne},"
                + $"сторона two: {SideTwo}, "
                + $"периметр: {Perimeter()}, "
                + $"площа: {Area()}, "
                + $"колір: {Color.GetColor()}";
        }
    }
}
